Content.Shared/Research/Prototypes/LatheRecipePrototype.cs

```diff
@@ -1,101 +1,58 @@
         /// </summary>
-        [DataField("result", required: true, customTypeSerializer:typeof(PrototypeIdSerializer<EntityPrototype>))]
-        public string Result = string.Empty;
+        [DataField]
+        public EntProtoId? Result;
+
+        [DataField]
+        public Dictionary<ProtoId<ReagentPrototype>, FixedPoint2>? ResultReagents;
 
         /// <summary>
         ///     An entity whose sprite is displayed in the ui in place of the actual recipe result.
         /// </summary>
-        [DataField("icon")]
+        [DataField]
         public SpriteSpecifier? Icon;
 
         [DataField("completetime")]
-        private TimeSpan _completeTime = TimeSpan.FromSeconds(5);
-
-        [DataField("materials", customTypeSerializer: typeof(PrototypeIdDictionarySerializer<int, MaterialPrototype>))]
-        private Dictionary<string, int> _requiredMaterials = new();
-
-        //todo make these function calls because we're eating tons of resolves here.
-        /// <summary>
-        ///     Name displayed in the lathe GUI.
-        /// </summary>
-        [ViewVariables]
-        public string Name
-        {
-            get
-            {
-                if (_name.Trim().Length != 0)
-                    return _name;
-                var protoMan = IoCManager.Resolve<IPrototypeManager>();
-                protoMan.TryIndex(Result, out EntityPrototype? prototype);
-                if (prototype?.Name != null)
-                    _name = prototype.Name;
-                return _name;
-            }
-        }
-
-        /// <summary>
-        ///     Short description displayed in the lathe GUI.
-        /// </summary>
-        [ViewVariables]
-        public string Description
-        {
-            get
-            {
-                if (_description.Trim().Length != 0)
-                    return _description;
-                var protoMan = IoCManager.Resolve<IPrototypeManager>();
-                protoMan.TryIndex(Result, out EntityPrototype? prototype);
-                if (prototype?.Description != null)
-                    _description = prototype.Description;
-                return _description;
-            }
-        }
+        public TimeSpan CompleteTime = TimeSpan.FromSeconds(5);
 
         /// <summary>
         ///     The materials required to produce this recipe.
         ///     Takes a material ID as string.
         /// </summary>
-        [ViewVariables]
-        public Dictionary<string, int> RequiredMaterials
-        {
-            get => _requiredMaterials;
-            private set => _requiredMaterials = value;
-        }
-
-
-        /// <summary>
-        ///     How many milliseconds it'll take for the lathe to finish this recipe.
-        ///     Might lower depending on the lathe's upgrade level.
-        /// </summary>
-        [ViewVariables]
-        public TimeSpan CompleteTime => _completeTime;
+        [DataField]
+        public Dictionary<ProtoId<MaterialPrototype>, int> Materials = new();
```

Comment:
Why was this renamed?

Reply:
the general convention nowadays is to have variables match the datafield string exactly. its a relatively minor change and doesn't affect any yaml so i thought it'd just be good to update while i was working with everything else.